lib/valkyrie/persistence/postgres/resource_converter.rb

```diff
@@ -9,7 +9,7 @@ def initialize(resource, resource_factory:)
     end
 
     def convert!
-      orm_class.find_or_initialize_by(id: resource.id.to_s).tap do |orm_object|
+      orm_class.find_or_initialize_by(id: resource.id && resource.id.to_s).tap do |orm_object|
```

Comment:
Why do we need the cast?

Reply:
I had problems when the pg database's id type is string and the resource you are persisting has no id defined (ie expectation is that postgres would populate it). nil.to_s converts to "", which is a valid string, so pg would not invoke the default for the field. Had to change it to only convert if id is present so that when it generated the insert statement it would leave id as null so that the database default can be generated.